### ISSUE TITLE
Release bot as GitHub Action

### ISSUE THREAD

**Issue by CermakM** [user-cont/release-bot#251]
Hello,

I've been doing some groundwork around this. I am creating an issue so that I could link PRs and additional resources for this one.

Intro:
The GitHub actions are quite a new feature that GitHub opened for beta some time back. Check out [the article I wrote about it](https://medium.com/@marekermk/https-medium-com-marekermk-guide-to-a-custom-ci-cd-with-github-actions-5aa0ff07a656).

Justification of the idea:
In comparison to GitHub Applications (which might seem like a similar approach and there already is an [issue](#237) for that), there are significant differences:

| - | App | Action |
| - | ----- | --------- |
| Token | Has to be provided by user, encoded / secured and provided to the application | autogenerated by GitHub |
| Server | Has to be hosted | Provided by GitHub |
| Implementation | Custom server, load balancing, queuing and routing is required | Runs on GitHub hosted machine |
| Customizability | Configuration has to be present in the repo | Customizable via arguments, Release bot configuration is autogenerated

Disadvantages: Slower response time on events since the Action has to pull the image first. Approximately 1 minute. Given that release is not a thing that would typically require quick interaction, the impact of this is quite low.

Current implementation in progress: https://github.com/CermakM/python-release-action

**Comment by rpitonak** [comment]
Hi Marek,

Thanks for your effort on this one. As far as I know, we can support both the app and the action, right?

I definitely like the idea of the release-bot as Github action and I am curious how the PyPI token would be handled in this case?

**Comment by rpitonak** [comment]
Thank you for your answer, but my question was actually about PyPI token since release-bot can handle also PyPI releases.

**Comment by CermakM** [comment]
Ah right, sorry, somehow miss the PyPI part -- there is such thing as PyPI token? :open_mouth:

**Comment by rpitonak** [comment]
I meant `.pypirc` file with user credentials.

**Comment by CermakM** [comment]
Correct, ye ... The current implementation of the GH Action allows to feed the yaml configuration as an argument to that action, see the [action.yml](https://github.com/CermakM/python-release-action/blob/master/action.yml) for more info.
